apps/charterafrica/src/components/Datasets/Datasets.js

```diff
@@ -0,0 +1,128 @@
+import DatasetFilterBar from "./DatasetFilterBar";
+import useDatasets from "./useDatasets";
+
+import NextPrevPagination from "@/charterafrica/components/NextPrevPagination";
+import queryString from "@/charterafrica/utils/datasets/queryString";
+
+const Datasets = React.forwardRef(function Datasets(
+  {
+    sx,
+    data: datasetsProp,
+    labels,
+    tags = [],
+    countries = [],
+    totalPages: originalTotalPages,
+    sortOptions = [],
+  },
+  ref
+) {
+  const [page, setPage] = useState(1);
+  const [q, setQ] = useState("");
+  const [sort, setSort] = useState("");
+  const [selectedCountries, setSelectedCountries] = useState([]);
+  const [selectedTags, setSelectedTags] = useState([]);
+  const [totalPages, setTotalPages] = useState(originalTotalPages);
+  const [datasets, setDatasets] = useState(datasetsProp || []);
+  const router = useRouter();
+  const datasetsRef = useRef();
+  useImperativeHandle(ref, () => datasetsRef.current);
+  const { asPath } = router;
+
+  const handleChangePage = (_, value) => setPage(value);
+  const handleChangeQ = (_, value) => setQ(value);
+  const handleChangeSort = (_, value) => setSort(value);
+  const handleChangeCountries = (_, value) => {
+    setSelectedCountries(value);
+  };
+
+  const handleChangeTags = (_, value) => setSelectedTags(value);
+
+  const query = queryString({
+    page,
+    q,
+    sort,
+    countries: selectedCountries,
+    tags: selectedTags,
+  });
+
+  useEffect(() => {
+    const pathname = asPath.split("?")[0];
+    router.push({ pathname, query }, undefined, {
+      scroll: false,
+      shallow: true,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query]);
+
+  const { data, isLoading } = useDatasets({
+    q,
+    sort,
+    page,
+    countries: selectedCountries,
+    tags: selectedTags,
+  });
+
+  if (isLoading && datasetsRef.current) {
+    datasetsRef.current.scrollIntoView({ behavior: "smooth" });
+  }
+
+  useEffect(() => {
+    if (!isLoading) {
+      const { datasets: filteredDatasets, totalPages: filteredTotalPages } =
+        data || {};
+      setDatasets(filteredDatasets);
+      setTotalPages(filteredTotalPages);
+    }
+  }, [data, isLoading]);
+
+  return (
+    <Box
+      bgcolor="common.white"
+      sx={{
+        scrollMarginTop: { xs: "56px", sm: "64", md: "114px" },
+        ...sx,
+      }}
+      ref={datasetsRef}
+    >
+      <Section
+        sx={{ px: { xs: 2.5, sm: 0 }, py: { xs: 5, md: 0 }, pb: { md: 5 } }}
+      >
+        <DatasetFilterBar
+          tags={tags}
+          countries={countries}
+          labels={labels}
+          sortOptions={sortOptions}
+          onQChange={handleChangeQ}
+          onSortChange={handleChangeSort}
+          onChangeCountries={handleChangeCountries}
+          onChangeTags={handleChangeTags}
+        />
+        {isLoading ? <LinearProgress color="secondary" /> : null}
+        {datasets?.map((dataset) => (
+          <DatasetCard
```

Comment:
Since DatasetCard has a border, this way, 2 DatasetCards next to each other are going to have 2 borders: border-bottom of top card and border-top of bottom card.

You need to but these cards inside say a MUI `Stack` and hide on of the borders e.g. show only top border for first item but hide for the rest, etc., using css selectors `:first-child` or :first-of-type